x-pack/plugins/security/server/routes/authorization/roles/put.ts

```diff
@@ -14,7 +16,40 @@ import {
   transformPutPayloadToElasticsearchRole,
 } from './model';
 
-export function definePutRolesRoutes({ router, authz, clusterClient }: RouteDefinitionParams) {
+const roleGrantsSubFeaturePrivileges = (features: Feature[], role: Role) => {
+  const subFeaturePrivileges: Record<string, SubFeaturePrivilegeConfig[]> = features.reduce(
+    (acc, feature) => {
+      return {
+        ...acc,
+        [feature.id]: feature.subFeatures
+          ?.map((sf) => sf.privilegeGroups.map((pg) => pg.privileges))
+          .flat(2),
+      };
+    },
+    {}
+  );
+
+  const hasAnySubFeaturePrivileges = (role.kibana ?? []).some((kibanaPrivilege) =>
```

Comment:
__question:__ am I understanding this correctly that we're recording not the fact that subfeature privileges are used per se, but rather the fact that user chose to _customize_ sub features (e.g. when user chooses `All`, all subfeature privileges are "granted" and relied upon implicitly if I'm not missing anything)? If so, wondering if we should make that clear in the feature usage name/id (e.g. put `custom` in there or something like this). Not important at all, just curious.

Reply:
That's exactly right. We are only recording the fact that a role was created/updated with explicit sub-feature privileges granted, as opposed to the "primary" feature privileges, or the base privileges.

I think the usage name is ok for the time being. The _intent_ is to figure out when sub-feature privileges are being used at all, and this customization check is a "close enough" approximation of this. There is potential that we'll change the way we check for this in the future.

Reply:
Sure, works for me too :+1: